subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/FlakinessIssueReporter.groovy

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.results
+
+import groovy.transform.CompileStatic
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.gradle.ci.common.model.FlakyTest
+import org.gradle.ci.github.GitHubIssuesClient
+import org.gradle.ci.tagging.flaky.KnownFlakyTestProvider
+import org.gradle.performance.results.ScenarioBuildResultData.ExecutionData
+import org.kohsuke.github.GHIssue
+import org.kohsuke.github.GHIssueState
+
+import static org.gradle.ci.github.GitHubIssuesClient.CI_TRACKED_FLAKINESS_LABEL
+import static org.gradle.ci.github.GitHubIssuesClient.FROM_BOT_PREFIX
+import static org.gradle.ci.github.GitHubIssuesClient.MESSAGE_PREFIX
+import static org.gradle.ci.github.GitHubIssuesClient.TEST_NAME_PREFIX
+
+@CompileStatic
+class FlakinessIssueReporter {
+    static final String GITHUB_FIX_IT_LABEL = "fix-it"
+    private final KnownFlakyTestProvider provider
+    private final GitHubIssuesClient gitHubIssuesClient
+
+    FlakinessIssueReporter(GitHubIssuesClient gitHubIssuesClient, KnownFlakyTestProvider provider) {
+        this.provider = provider
+        this.gitHubIssuesClient = gitHubIssuesClient
+    }
+
+    void report(ScenarioBuildResultData flakyScenario) {
+        FlakyTest knownFlakyTest = findKnownFlakyTest(flakyScenario)
+        if (knownFlakyTest) {
+            if (issueClosed(knownFlakyTest)) {
+                knownFlakyTest.issue.reopen()
+            }
+            if (!hasFixItLabel(knownFlakyTest)) {
+                knownFlakyTest.issue.addLabels(GITHUB_FIX_IT_LABEL)
+            }
+        } else {
+            knownFlakyTest = openNewFlakyTestIssue(flakyScenario)
+        }
+
+        commentCurrentFailureToIssue(flakyScenario, knownFlakyTest.issue)
+    }
+
+    @TypeChecked(TypeCheckingMode.SKIP)
+    static void commentCurrentFailureToIssue(ScenarioBuildResultData scenario, GHIssue issue) {
+        issue.comment("""
+${FROM_BOT_PREFIX}
+
+Coordinator url: https://builds.gradle.org/viewLog.html?buildId=${System.getenv("BUILD_ID")}
+Worker url: ${scenario.webUrl}
+Agent: [${scenario.agentName}](${scenario.agentUrl})
+Details:
+
+| Iteration | Difference | Confidence |
+|---|---|---|
+${assembleTable(scenario)}
+""")
+    }
+
+    private static String assembleTable(ScenarioBuildResultData scenario) {
+        scenario.executions.withIndex().collect { ExecutionData execution, int index ->
+            "|${index + 1}|${execution.differenceDisplay}|${execution.formattedConfidence}|"
+        }.join('\n')
+    }
+
+    private FlakyTest openNewFlakyTestIssue(ScenarioBuildResultData flakyScenario) {
+        String testName = flakyScenario.scenarioName
```

Comment:
Shall we use the scenario name or the full test name, something like `org.gradle.performance.regression.buildcache.TaskOutputCachingJavaPerformanceTest.clean assemble for non-abi change on largeJavaMultiProject with local cache (parallel: true)` ?

Reply:
In rare cases the scenario name would conflict, but I don't like long FQCN, so I'd say let's use simpleName + scenarioName `TaskOutputCachingJavaPerformanceTest.clean assemble for non-abi change on largeJavaMultiProject with local cache (parallel: true)`

Reply:
I would do the same as we do for the other test flakiness issues.

Reply:
What do you think if we change this behavior for other flakiness issues? Because currently I always change the FQCN in flakiness issue to simple name to make it more readable.

Reply:
Done with FQCN.